File: src/main/java/ai/devchat/common/Log.java
package ai.devchat.common;

import com.intellij.openapi.diagnostic.LogLevel;
import com.intellij.openapi.diagnostic.Logger;

import ai.devchat.cli.DevChatInstallationManager;

public class Log {
    private static final Logger LOG = Logger.getInstance(DevChatInstallationManager.class);
    private static final String PREFIX = "[DevChat] ";

    private static void setLevel(LogLevel level) {
        LOG.setLevel(level);
    }

    public static void setLevelInfo() {
        LOG.setLevel(LogLevel.INFO);
    }

    public static void setLevelDebug() {
        LOG.setLevel(LogLevel.DEBUG);
    }

    public static void info(String message) {
        LOG.info(PREFIX + message);
    }

    public static void error(String message) {
        LOG.error(PREFIX + message);
    }

    public static void debug(String message) {
        LOG.debug(PREFIX + message);
    }
}

File: src/main/kotlin/ai/devchat/common/Log.kt
package ai.devchat.common

import ai.devchat.common.Constants.ASSISTANT_NAME_ZH
import com.intellij.openapi.diagnostic.LogLevel
import com.intellij.openapi.diagnostic.Logger

object Log {
    private val LOG = Logger.getInstance(ASSISTANT_NAME_ZH)
    private fun setLevel(level: LogLevel) {
        LOG.setLevel(level)
    }

    @JvmStatic
    fun setLevelInfo() {
        LOG.setLevel(LogLevel.INFO)
    }

    fun setLevelDebug() {
        LOG.setLevel(LogLevel.DEBUG)
    }

    @JvmStatic
    fun info(message: String) {
        LOG.info(message)
    }

    @JvmStatic
    fun error(message: String) {
        LOG.error(message)
    }

    @JvmStatic
    fun warn(message: String) {
        LOG.warn(message)
    }

    fun debug(message: String) {
        LOG.debug(message)
    }
}
